Modules/Module_Processing/Module_ASL/xASL_wrp_ProcessM0.m

```diff
@@ -120,10 +131,40 @@ function xASL_wrp_ProcessM0(x)
 
 			% Find the smallest of the previously selected positive nonzero TEs
 			minTE = min(uniqueNonzeroTE);
-			fprintf('Multi-TE M0 present. Using the shortest TE=%.1f ms for M0 calibration.\n', minTE*1000);
+			% JSON values were converted from BIDS (seconds) to Legacy (millisecond) during JSON reading
+			fprintf('Multi-TE M0 present. Using the shortest TE = %.1f ms for M0 calibration.\n', minTE);

```

Comment:
Why is this not a warning and 0D has a warning?

Reply:
Cause with multi-TE, it's kind of assumed that you only take the first TE with best signal as it has the same TE as the volumes really used for CBF quantification. So there's no logical reason to take other TEs. So this comments only documents that we did that.

For TR, that's more tricky. We could pick another TR as more optimal (or some people might prefer to), or we could run an inversion recovery fit. So this solution is not necessarily the only and the best, so we have to warn the user we did a choice for him...